### ISSUE TITLE
[no-unresolved] False positive with the module `gecko-profiler-demangle` (which embeds a wasm module)

### ISSUE THREAD

**Issue by julienw** [benmosher/eslint-plugin-import#2098]
Hey,

I got this new issue recently, because we use async import to import this module in our project, but I suspect this happens from previous versions too.

I made this testcase project to help debugging, with a simpler usage:

```
git clone https://github.com/julienw/bug-eslint-import-wasm
cd bug-eslint-import-wasm
npm ci
npm run lint
```
=>
```
test-eslint-import/index.js
  1:30  error  Unable to resolve path to module 'gecko-profiler-demangle'  import/no-unresolved
```

This testcase is minimal but this is used in a bigger project: https://github.com/firefox-devtools/profiler/blob/acbf66a48233811cc5499604de48fa790ddbe672/src/profile-logic/symbol-store.js#L150-L151

The problem only became apparent with the latest update with async import support.

This project uses webpack v4, so it's totally possible that webpack does some magic and some configuration is missing with eslint or the plugin, but it's not clear to me, so I prefered filing this bug first.

The source of this module is in https://github.com/mstange/gecko-profiler-demangle and the npm package is in https://www.npmjs.com/package/gecko-profiler-demangle. I think it's generated using [wasm-pack](https://github.com/rustwasm/wasm-pack), but probably an old version of it.
Looking at my node_modules it looks reasonable but I don't have much knowledge around how node_modules works. I'm sure you'll know better :-)

Thanks!

**Comment by ljharb** [comment]
Looking at https://unpkg.com/browse/gecko-profiler-demangle@0.2.0/package.json, it doesn't specify a "main" or "exports", nor does it have an "index.js", so it is in fact a module that can't be required or imported - the error is correct, and "webpack magic" is likely the reason things work at all.

The package needs fixing - it should use "exports", and should point at https://unpkg.com/browse/gecko-profiler-demangle@0.2.0/gecko_profiler_demangle_bg.wasm under some kind of custom condition (probably "wasm") and https://unpkg.com/browse/gecko-profiler-demangle@0.2.0/gecko_profiler_demangle.js under the "default" condition.

In the meantime, you can configure the plugin to ignore this module with the [`import/ignore` setting](https://github.com/benmosher/eslint-plugin-import#importignore).
